sdk/servicebus/azure-servicebus/api_review.py

```diff
@@ -0,0 +1,588 @@
+    def receive(
+        self,
+        max_batch_size : int = None,
+        timeout : float = None,
+    ) -> List[Message]:  # Pull mode receive
+    def settle_deferred_messages(
+        self,
+        settlement : str,  # TODO: In T1 settlement is just string like 'completed', 'suspended', 'abandoned', can we improve this parameter to be Enum or remove this method?
+        messages : List[DeferredMessage],
+        **kwargs
+    ) -> None:  # Batch settle deferred messages
+
+    def list_sessions(self) -> List[str]:
+    def get_session(self) -> Session:  # raise Error when called on non-session entity
+
+    # Rule APIs, raise Error when called on Queue
+    def add_rule(self, rule_name : str, filter : str) -> None: # TODO: figure out what filter is
+    def remove_rule(self, rule_name : str) -> None:
+    def get_rules(self) -> List[str] -> Dict[str, Any]:
+
+
+class Session:
+    def get_session_id(self) -> str:
+    def get_session_state(self) -> str:  # This can be made to property if property is more pythonic way
+    def set_session_state(self, state : str) -> None:  # This can be made to property if property is more pythonic way
+    def renew_lock(self) -> None:
+    @property
+    def expired(self) -> bool:
+
+
+class Transaction:
+    def __init__(self) -> None:
+    def __enter__(self):
+    def __exit__(self):
+    def begin(self) -> None:
+    def commit(self) -> None:
+    def abort(self) -> None:
+
+
+class Message:
+    def __init__(self, body : str, encoding : str = 'UTF-8', **kwargs) -> None:
+    def __str__(self):
+
+    # @properties
+    def settled(self) -> bool:  # read-only
+    def enqueued_time(self) -> datetime:  # read-only
+    def scheduled_enqueue_time(self) -> datetime:  # read-only
+    def sequence_number(self) -> int:  # read-only
+    def partition_id(self) -> str:  # read-only
+    def locked_until(self) -> datetime:  # read-only
+    def expired(self) -> bool:  # read-only
+    def lock_token(self) -> str:  # read-only
+    def body(self) -> Union[bytes, Generator[bytes]]:  # read-only
+    def partition_key(self, value : str):
+    def partition_key(self) -> str:
+    def via_partition_key(self, value: str):
+    def via_partition_key(self) -> str:
+    def sessio_id(self, value : str):
+    def sessio_id(self) -> str:
+    def time_to_live(self, value : Union[float, timedelta]):
+    def time_to_live(self) -> datetime.timedelta:
+    def annotations(self, value : dict[str, Any]):
+    def annotations(self) -> dict[str, Any]:
+    def user_properties(self, value : dict[str, Any]):
+    def user_properties(self) -> dict[str, Any]:
+    def enqueue_sequence_number(self, value : int):
+    def enqueue_sequence_number(self) -> int:
+
+    # Methods
+    def schedule(self, schedule_time : datetime) -> None:
+    def renew_lock(self) -> None:
+    def complete(self) -> None:
+    def dead_letter(self, description : str = None) -> None:
+    def abandon(self) -> None:
+    def defer(self) -> None:
```

Comment:
Likewise, what's the behaviour for all these functions for an outgoing message?

Reply:
We should raise error.

From this point of view, these methods looks more reasonable if they stay on the `ReceiverClient`.

Another option is that we have two type of `Message`: `IncomingMessage` and `OutgoingMessage`.

Reply:
as per discussion, having a separate class `ReceivedMessage` to hold these settlement methods.